### PR TITLE
Cleanup obsoletions for V5

### PR DESCRIPTION
**Pull request by MaceWindu**
- Removed APIs, obsoleted in v4:
  - some `UseOracle()` configuration method overloads
  - `DataConnection.OnTrace` property
  - some `OracleDataProvider` constructors
  - `OracleTools`:
    - `DetectedProviderName` property
    - `GetDataProvider` method (one of overloads)
    - `CreateDataConnection` method (some overloads)
    - `DetectProviderName` method
    - `GetVersionedDataProvider` method
    - `ResolveOracle` methods
  - `SqlServerTools.Sql` class
  - `AssociationAttribute`: `KeyName`, `BackReferenceName`, `IsBackReference`, `Relationship` properties
  - `Relationship` enum
  - `SqlDataType.GetDataType(Type)` overload
- ~Obsoleted~ removed some internal APIs, related to removed `SqlDataType.GetDataType(Type)` method:
  - `SqlDataType(Type)` constructor
  - following `ValueToSqlConverter` methods (added new methods with additional `MappingSchema` parameter):
    - `bool TryConvert(StringBuilder stringBuilder, object? value)`
    - `bool TryConvert(StringBuilder stringBuilder, SqlDataType? dataType, object? value)`
    - `StringBuilder Convert(StringBuilder stringBuilder, object? value)`
    - `StringBuilder Convert(StringBuilder stringBuilder, SqlDataType? dataType, object? value)`
- fix typo in `BasicSqlOptimizer.ConvertConvertion` method name
- `SqlTable` indexer replaced with `FindFieldByMemberName` method

### CODE REVIEW

File: Source/LinqToDB/SqlProvider/ValueToSqlConverter.cs [hunk 1]
```diff
@@ -166,11 +167,13 @@ static void BuildDateOnly(StringBuilder stringBuilder, DateOnly value)
 		}
 #endif
 
+		[Obsolete($"Use overload with MappingSchema parameter")]
```
Comment (viceroypenguin) [inline Source/LinqToDB/SqlProvider/ValueToSqlConverter.cs:170]: Doesn't need interpolated string. Should devolve to same code, but unnecessary claim. (+2 more in file)

File: Source/LinqToDB/SqlQuery/SqlTable.cs [hunk 2]
```diff
@@ -169,18 +169,14 @@ public override string ToString()
...
+		/// <param name="memberName">Mapping class member name.</param>
+		public SqlField? FindFieldByMemberName(string memberName)
 		{
-			get
-			{
-				_fieldsLookup.TryGetValue(fieldName, out var field);
-				return field;
-			}
+			_fieldsLookup.TryGetValue(memberName, out var field);
+			return field;
```
Comment (viceroypenguin) [inline Source/LinqToDB/SqlQuery/SqlTable.cs:179]: This change (and all related) is large enough to probably belong in separate related PR?

**Review by viceroypenguin — APPROVED** [review]

**Review by sdanyliv — APPROVED** [review]

**Review by viceroypenguin — APPROVED** [review]

**Pull request merged**